pkg/R/DataFrame.R

```diff
@@ -193,11 +363,62 @@ setMethod("take",
             collect(limited)
           })
 
+#' Head
+#'
+#' Return the first NUM rows of a DataFrame as a data.frame. If NUM is NULL, 
+#' then head() returns the first 6 rows in keeping with the current data.frame 
+#' convention in R.
+#'
+#' @param x A SparkSQL DataFrame
+#' @param num The number of rows to return. Default is 6.
+#' @return A data.frame
+#'
+#' @rdname head
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' head(df)
+#' }
+
+setMethod("head",
+          signature(x = "DataFrame"),
+          function(x, num = 6L) {
+          # Default num is 6L in keeping with R's data.frame convention
+            take(x, num)
+          })
+
+#' Return the first row of a DataFrame
+#'
+#' @param x A SparkSQL DataFrame
+#'
+#' @rdname first
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' first(df)
+#' }
+
+setGeneric("first", function(x) {standardGeneric("first") })
+
+setMethod("first",
+          signature(x = "DataFrame"),
+          function(x) {
+            take(x, 1)
```

Comment:
You can do that now. Since `first` gives you a `data.frame` with named columns, we could run your example as it's written and it would work. Here's an example from the console:
```R
> obj <- first(df)
> class(obj)
[1] "data.frame"
> obj$name == "Michael"
[1] TRUE
```
Does that work?